modules/contributor-guide/partials/proc_adding-a-che-theia-plug-in-into-a-workspace.adoc

```diff
@@ -7,67 +7,64 @@
 
-. Navigate to the *Workspaces* tab in the dashboard.
-. Select the workspace to which you want to add the plug-in.
-. In *Workspace > _<workspace-name>_* window, click the *Plugins* tab, and select the required plug-ins.
+. Run the `Deploy Plugin by Id` command, and specify the ID of the plug-in that you want to add to the workspace. A plug-in ID can be:
++
+** A local directory with plug-ins. All the plug-in binaries (`.theia` files) in the specified directory would be loaded. Example: `local-dir:///home/theia/plugins/`.
+
+** VS Code extension link. To obtain the link, go to the VS Code marketplace, find the needed extension, copy its ID, and append the ID to the `vscode:extension/` prefix.
 +
-.Adding plug-ins through the dashboard
-image::extensibility/adding-plug-ins-in-dashboard.png[link="../_images/extensibility/adding-plug-ins-in-dashboard.png"]
+For example, for the link:https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml[XML language server] plug-in, it is: `vscode:extension/redhat.vscode-xml`.
 
+** Link to plug-in binaries (`.theia` files). Can be both HTTP and HTTPS. Example: `https://domain.net/path/plugin.theia`.
+
+To check if the plug-in loaded successful, open the *Plugins panel* (*View* > *Plugins* in the main menu) and search for the plug-in in the list.
 
 [id="adding-a-plug-in-by-configuring-a-workspace_{context}"]
-== Adding a plug-in by configuring a devfile
+== Adding a plug-in by configuring a workspace devfile
 
 This method is useful when you need to add plug-ins from a non-default registry.
 
 . Navigate to the *Workspaces* tab in the dashboard.
 . Select the workspace to which you want to add the plug-in.
 . Switch to the *Devfile* tab, and find the `components` section.
-+
-.Adding plug-ins by configuring a component
-image::extensibility/adding-plug-ins-in-workspace-devfile.png[link="../_images/extensibility/adding-plug-ins-in-workspace-devfile.png"]
-
-+
-Plug-ins are added in YAML format as array elements in the `components` section.
-. To add plug-ins, append this lines to the `components` section. A record has the following format:
+. To add plug-ins, append your plugin component to the `components` section. A record has the following format:
 +
 ----
+components:
   - type: chePlugin
     reference: >-
-      <registry-base-uri>/v3/plugins/<plugins-group>/<plugin-name>/<version>
+      <registry-base-uri>/v3/plugins/<plugin-publisher>/<plugin-name>/<version>
 ----
 +
 Reference URL examples:
 +
 * pass:[https://custom-registry/v3/plugins/custom-plugin-name/0.0.1/]
 * pass:[https://raw.githubusercontent.com/username/che-custom-plugins/master/v3/plugins/custom-plugin-name/0.0.1/meta.yaml]
 
+[id="sharing-a-plug-in-with-factory-url_{context}"]
+== By preparing shareable Devfile
```

Comment:
This section is mostly copy/paste from removed factory section where I replaces removed persisted factories with che-host/f?url=... link.
For me, it does not seem detailed and clear enough but I don't have any experience with writing plugins and not sure how it can be made better.